### PR TITLE
Add tunable HyperMinHash impl

### PR DESCRIPTION
**Pull request by sherifnada**
This PR cleans up a lot of the repo and adds `HyperMinHash`, an implementation that is based on `HyperLogLog` with the addition of bias correction from `HyperLogLog++`.

### CODE REVIEW

File: src/main/java/com/liveramp/hyperminhash/BetaMinHash.java [hunk 1]
```diff
@@ -1,157 +1,182 @@
...
+  public boolean offer(byte[] val) {
+    MetroHash128 hash = new MetroHash128(HASH_SEED).apply(ByteBuffer.wrap(val));
+    ByteBuffer buf = ByteBuffer.allocate(16);
+    hash.writeBigEndian(buf);
+    return addHash(buf);
   }
 
-  public long cardinality() {
-    return BetaMinHashCardinalityGetter.cardinality(this);
+//  @Override
```
Comment (ChristianHansen) [inline src/main/java/com/liveramp/hyperminhash/BetaMinHash.java]: Need to remove this. We shouldn't have commented-out code checked in

File: src/main/java/com/liveramp/hyperminhash/HyperMinHash.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  }
+
+  @Override
+  public long cardinality() {
+    return HmhCardinalityEstimator.estimateCardinality(registers, p, r);
+  }
+
+  @Override
+  public boolean offer(byte[] bytes) {
+//    MetroHash128 hash = MetroHash.hash128(HASH_SEED, bytes);
```
Comment (ChristianHansen) [inline src/main/java/com/liveramp/hyperminhash/HyperMinHash.java]: We should remove this

File: src/main/java/com/liveramp/hyperminhash/BetaMinHash.java (hunk 1 above)
Comment (ChristianHansen) [reply src/main/java/com/liveramp/hyperminhash/BetaMinHash.java]: Need to implement a `SerDe` class instead.

Comment (sherifnada) [reply src/main/java/com/liveramp/hyperminhash/BetaMinHash.java]: replaced with `BetaMinHashSerde`

**Pull request merged**